libraries/GCS_MAVLink/GCS_MAVLink.cpp

```diff
@@ -110,15 +111,30 @@ void comm_send_buffer(mavlink_channel_t chan, const uint8_t *buf, uint8_t len)
 /*
   lock a channel for send
  */
-void comm_send_lock(mavlink_channel_t chan)
+void comm_send_lock(mavlink_channel_t chan_m, uint16_t size)
 {
-    chan_locks[(uint8_t)chan].take_blocking();
+    const uint8_t chan = uint8_t(chan_m);
+    chan_locks[chan].take_blocking();
+    if (mavlink_comm_port[chan]->txspace() < size) {
+        chan_discard[chan] = true;
+    }
 }
 
 /*
   unlock a channel
  */
-void comm_send_unlock(mavlink_channel_t chan)
+void comm_send_unlock(mavlink_channel_t chan_m)
+{
+    const uint8_t chan = uint8_t(chan_m);
```

Comment:
This isn't doing the idomatic `-MAVLINK_COMM_0` thing - perhaps we should just give up on that and always assume it's zero now-adays.

Reply:
i think we can